plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlSource.java

```diff
@@ -16,7 +18,7 @@
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 
-public class YamlSource<T> {
+public class YamlSource<T> implements Closeable {
```

Comment:
Implement autocloseable? then we could probably use try-with-resources?

Reply:
How would you write a try catch block for a list?

Reply:
ugh

Reply:
I can spent more time rewriting all the code just to enable try with resources for a list of yaml sources.

Reply:
btw also looking through the call stack, it would be a nightmare trying to rewrite the code to.

Since there is a lot of checkWith and configureWith calling.

Lucky loooking there yamlSources go through these two closeSources should do the trick.

Reply:
could still implement auto-closeable for later ? you don't have to use it